app/models/concerns/presenters/statemachine/feature_in_states.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+module Presenters::Statemachine
+  # This module provides a method to determine if a feature can be enabled
+  # based on the current state of the instance.
+  #
+  # State names:
+  # Presenters::StandardPresenter.state_machines[:state].states.map(&:value)
+  # pending, passed, started, cancelled, failed, processed_1, processed_2,
+  # processed_3, processed_4, qc_complete, unknown
+  module FeatureInStates
+    extend ActiveSupport::Concern
+
+    included do
+      # Determines if a feature can be enabled based on provided states.
+      #
+      # @param states [String, Symbol, Array, Hash] The states to check against.
+      # This can be a single state as a String or Symbol, an Array of states,
+      # or a Hash with includes and excludes keys for states.
+      # @param default [Boolean] The default return value if no specific
+      # condition is met.
+      # @return [Boolean] Returns true if the feature can be enabled based on
+      # the states parameter, otherwise returns the default.
+      #
+      # :reek:BooleanParameter
+      # :reek:ManualDispatch
+      # :reek:TooManyStatements
+      def can_be_enabled?(states, default: true)
+        return default unless respond_to?(:state) # No state method.
+        return default if states.blank? # No states to check.
+
+        current_state = state.to_s
+        return default if current_state.blank? # No current state.
+
+        includes, excludes = parse_states(states)
+
+        # Exludes take precedence over includes.
+        return false if excludes.present? && excludes.include?(current_state)
+
+        # If includes is present, the current state must be in the list.
+        includes.present? ? includes.include?(current_state) : default
+      end
+
+      private
+
+      # Parses the states parameter into includes and excludes arrays.
+      #
+      # @param states [String, Symbol, Array, Hash] The states to check against.
+      # @return [Array] Returns an array with two elements: includes and excludes.
+      #
+      # :reek:TooManyStatements
+      def parse_states(states)
+        case states
+        when Hash
+          states = states.stringify_keys
+          includes = parse_tokens(states['includes'])
+          excludes = parse_tokens(states['excludes'])
+        else
+          includes = parse_tokens(states)
+          excludes = []
+        end
+        [includes, excludes]
+      end
+
+      # Parses the state name tokens into an array of strings.
+      #
+      # @param tokens [String, Symbol, Array] The tokens to parse.
+      # @return [Array] Returns an array of strings.
+      #
+      # :reek:UtilityFunction { public_methods_only: true}
+      def parse_tokens(tokens)
+        case tokens
+        when String, Symbol
+          [tokens.to_s] # single
+        when Array
+          tokens.map(&:to_s) # multiple
+        else
+          [] # none
+        end
+      end
```

Comment:
Just opinion - I find this method a bit tricky to read - perhaps a bit less abstraction might be easier here?

Reply:
I gave Copilot this function and asked to optimise it. This is what it came up with:

```ruby
def parse_tokens(tokens)
  Array(tokens).map(&:to_s).compact
end
```

Did the same in [GPT-4o](https://openai.com/index/gpt-4o-and-more-tools-to-chatgpt-free/). It gave me this: 

```ruby
def parse_tokens(tokens)
  Array(tokens).map(&:to_s)
rescue NoMethodError
  []
end
```

Both are wrong 😆

Reply:
It is a simple method to accept symbol, string, or array VALUE in cases:
states: VALUE
In this case, it will be assumed that only the state name inclusions are specified

includes: VALUE 
The inclusions are specified separately under states. The method will collect the state names for inclusions

excludes: VALUE
The exclusions are specified separately under states. The method will collect the state names for exclusions

If there is no such value, it also collects an empty array. 

The method was named as parse_tokens to mean that state name words are collected whatever form they are given. Because the method name sounds too abstract and there is not enough comment to explain the details, it was responded with a code-review comment.

I will change the name and add comments. The method is just so useful. I won't change or throw away.